services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/RuleJobServiceImpl.java

```diff
@@ -92,7 +93,7 @@ public ResponseEntity<Object> runRuleJob(RuleJobRequest ruleJobRequest, int user
             dbUtil.changeSchema(project.getDbSchemaName());
             for (Integer ruleId : ruleJobRequest.getRuleIds()) {
 
-                QsRuleJob ruleJob = ruleJobRepository.findByRuleId(ruleId);
+                QsRuleJob ruleJob = ruleJobRepository.findByRuleIdAndActiveIsTrue(ruleId);
```

Comment:
response.put("code", HttpStatus.SC_BAD_REQUEST); - if multiple rule is having issues , then this will overwrite previous rule id's code and message..